wp_api/src/login/url_discovery.rs

```diff
@@ -411,47 +433,53 @@ pub enum FetchAndParseApiRootFailure {
     },
 }
 
-impl FetchAndParseApiRootFailure {
-    pub fn error_message(&self, parsed_site_url: impl Display) -> String {
+impl WpSupportsLocalization for FetchAndParseApiRootFailure {
+    fn message_bundle(&self) -> MessageBundle {
         match self {
-            Self::FetchApiRoot { error } => error.to_string(),
-            Self::ParseApiRoot {
-                parsing_error_message,
-            } => format!("Failed to parse api details: {:#?}", parsing_error_message),
-            Self::WpError {
-                error_code,
-                error_message,
-                status_code,
-            } => format!(
-                "WpError {{\n\tstatus_code: {}\n\terror_code: {:?}\n\terror_message: \"{}\"",
-                status_code, error_code, error_message
-            ),
+            Self::FetchApiRoot { error } => error.message_bundle(),
+            Self::ParseApiRoot { .. } => WpMessages::parse_api_root(),
+            Self::WpError { error_message, .. } => WpMessages::site_error_message(error_message),
             Self::ApplicationPasswordsNotSupported { reason, .. } => reason
                 .as_ref()
-                .map(|r| r.error_message(parsed_site_url))
-                .unwrap_or("Application Passwords are not supported".to_string()),
+                .map(|r| r.message_bundle())
+                .unwrap_or(WpMessages::application_passwords_not_supported()),
         }
     }
 }
 
-#[derive(Debug, Clone, uniffi::Error)]
+#[derive(Debug, Clone, uniffi::Error, WpDeriveLocalizable)]
 pub enum ApplicationPasswordsNotSupportedReason {
     ApplicationPasswordBlockedByPlugin {
+        site_url: Arc<ParsedUrl>,
```

Comment:
I specifically moved the common fields out of these sub error variants and this PR reverts that without any benefits. I'd like to stick to the single source of truth principle here which should be easy to do so by following the design in current `trunk`.

Also a little bit of general feedback; Making these types of changes as part of unrelated PRs can be easy to miss especially if they are not specifically brought up in the PR description.

Reply:
The existing implementation already showcases how this can be done, but I'll create a commit in another branch for you to cherry pick and comment here with the hash.

Reply:
You can cherry pick dbd841063e15372eb06475b44ca474f037e02c2c to address the issue.

Reply:
Yeap, it's OK. In fact, they should have been `uniffi::Enum`s, but the login process went through too many changes and we are left with some unintentional declarations.

These errors should never be thrown from any of the uniffi exported functions because they are only meaningful within the context of their outer errors. Using `uniffi::Enum` should help avoid accidentally throwing them. It's best to think of these as `reason`s for an error rather than being errors themselves.

Reply:
Their naming probably can be improved. There are `XxxError`, `XxxFailure`, and `XxxReason` types. I can see that the `XxxReason` types should not be thrown as errors, but the `XxxError` and `XxxFailure` seem pretty similar if we just look at their names.

Reply:
Agreed. I think we can use `XxxReason` as the naming scheme for these types. Feel free to make that change in this PR or I can open a new one once this is merged.

Reply:
I think this PR is large enough. If you don't mind, I'll merge this PR and leave the naming changes to you.

Reply:
Don't mind at all. In fact, I'd really like to merge this PR asap as what it has conflicts with what I am working on at the moment 😅